### DISCUSSION TITLE
OpenDTU mit Fritzbox verbunden aber nicht erreichbar

### DISCUSSION THREAD

**Discussion by Hamstihamsta** [tbnobody/OpenDTU#1074] [Q&A]
Moin moin,

ich versuche mich zum ersten Mal an einer OpenDTU. 

Gemäß Fritzbox Benutzeroberfläche ist die OpenDTU mit der Fritzbox verbunden. 
![Problem1](https://github.com/tbnobody/OpenDTU/assets/137622794/70b1739f-2653-4e8b-9411-454a1ebdf68d)

Wenn ich mich mit dem WLAN Netz OpenDTU verbinde, sehe ich bei "Netzwerkschnittstelle (Station) auch die IP-Adresse, unter der die OpenDTU eigentlich in WLAN Netz der Fritzbox erreichbar sein sollte. Ich kann sie jedoch nicht erreichen. 

Woran könnte das liegen? 

Fehlermeldung:
Die Webseite ist nicht erreichbar

http://192.169.... ist nicht erreichbar. 

ERR_ADDRESS_UNREACHABLE

Geflasht habe ich die OpenDTU heute, ist also auf dem aktuellen Stand. Verkabelung hab ich auch nochmal überprüft, passt. Ich war ja auch in der Lage das flashen durchzuführen und mich dann mit dem von der OpenDTU geöffneten WLAN Netzwerk zu verbinden und dort mein heimisches WLAN einzurichten.

**Comment by tbnobody** [comment]
https://avm.de/service/wissensdatenbank/dok/FRITZ-Box-7590/549_Datei-und-Druckerfreigaben-eines-Computers-sind-im-Heimnetz-nicht-sichtbar/

Siehe Punkt: Kommunikation zwischen WLAN-Geräten zulassen

Ggf. ist das relevant

**Reply by Hamstihamsta** [reply]
Ich habe eine Fritzbox 7490, dort ist das an einer anderen Stelle, aber:
Sowohl die Checkbox "Die unten angezeigten aktiven WLAN-Geräte dürfen untereinander kommunizieren" ist aktiviert, als auch die Checkbox "Alle neuen WLAN-Geräte zulassen" 

Aber: Bescheuerterweise funktioniert jetzt. Ohne dass ich irgendwas gemacht hab.. ich war beim Sport und alles war ausgeschaltet.. alles wieder an. Es geht. Ich versteh es nicht, will mich aber jetzt natürlich nicht beklagen. ;)

Danke! :)

**Reply by Wiechi73** [reply]
Moin! 
Ich habe dieses Phänomen ebenfalls an einer 7490. Bin lange auf 23.5.9 geblieben, damit hatte ich keine Probleme. Alle Versionen danach haben dieses Verhalten gezeigt. Seit 23.6.14 scheint es wieder besser zu laufen, mit 23.6.21 geht es auch. Das Eigenartige war, lokal über die IP nicht erreichbar, aber über eine Freigabe TCP Port über extern war das Webinterface sofort erreichbar. So konnte ich immer wieder downgraden per OTA. Seltsam…
Gruß Jens

**Comment by final-option** [comment]
Moin. 
Ich habe das gleiche Problem. 
DTU ist im WLAN, WEBGui lädt nicht. Es kommt allerdings keine Fehlermedlung, sondern im Browser (Chrome) einfach eine weiße Seite.
Die Einstellungen in der Fritzbox sind wie oben angegeben korrekt. 
Ich habe schon ein paar Mal den Stecker gezogen.
Wenn ich die DTU über die Powershell pinge bricht das nach 2-3 Werten mit einem Timeout ab (die DTU ist aber in der Fritzbox noch zu sehen).
Ich kann leider auch nicht mal die FW sehen, und der DTU AP ist ja auch weg nachdem das WLAN geht.
Per SSH kommt man anscheinend auch nicht auf die DTU?
Was tun?

Gruß

**Reply by final-option** [reply]
Hey,
ich habe jetzt den ESP neu geflasht, und dann die Einstellungen im DTU-WLAN angepasst. Also einmal die Zeit synchronisiert, die Fritzbox als Zeitserver eingetragen, den Wechselrichter eingerichtet, ein Update gemacht (vorher im eigenen WLAN die Datei runterladen!), die live Ansicht geprüft.
Und erst ganz am Schluß das WLAN eingerichtet.
Danach hat dann auch die WebGUI geladen und ich konnte normal auf die DTU zugreifen.
Keine Ahnung woran es am Ende gelegen hat. Aber vielleicht sollte man die Anleitung dahingehend anpassen, dass man alle Settings schon im DTU-WLAN vornimmt und dann erst auf das eigene WLAN wechselt?
Gruß

**Reply by tbnobody** [reply]
> Keine Ahnung woran es am Ende gelegen hat. Aber vielleicht sollte man die Anleitung dahingehend anpassen, dass man alle Settings schon im DTU-WLAN vornimmt und dann erst auf das eigene WLAN wechselt?

Macht eigentlich keinen Sinn und ist auch nicht so. Ich setze eine DTU ca. 1x pro Woche neu auf. Erster Schritt ist immer den ESP ins Wlan zu bringen und dann von dort aus diverse Einstellungen zu setzen. Klappt eigentlich immer zuverlässig.

**Reply by final-option** [reply]
Ich muss auch leider den Optimismus wieder zurücknehmen. Nach dem flashen am Laptop und dem Wechsel ins WLAN wurde mir in der Fritzbox die DTU mit 40Mbit angezeigt. 
Jetzt habe ich die DTU mit einem 5V/2A Netzteil im selben Zimmer wie der Router (vorher am Laptop im Nebenzimmer), und ich habe nur noch 3Mbit und die GUI lädt schon wieder nicht.

**Reply by tbnobody** [reply]
Naja, dann wieder am Laptop anschließen und gucken ob es wieder geht. Wenn ja --> Problem mit der Spannungsversorgung

**Reply by final-option** [reply]
Also mit 2 USB Netzteilen 5V/2A (1x NoName, 1x Xiaomi) funktioniert es nicht, keine GUI im Browser. Am Laptop habe ich zwar jetzt wieder knapp 20Mbit, aber die GUI lädt trotzdem nicht. 
Kann sich die DTU durch das Ab- und Anstecken der Stromversorgung irgendwie aufhängen?
Warum ging es vorhin und jetzt nicht mehr?

**Reply by tbnobody** [reply]
> Kann sich die DTU durch das Ab- und Anstecken der Stromversorgung irgendwie aufhängen?

Normalerweise nicht. Mir wäre nichts dergleichen bekannt.


> Warum ging es vorhin und jetzt nicht mehr?

Das kann ich nicht beantworten.

Siehst du etwas auf der Seriellen Konsole?

**Reply by final-option** [reply]
Wie komme ich auf die Konsole?

**Reply by tbnobody** [reply]
https://github.com/tbnobody/OpenDTU/wiki/FAQ#q-how-to-observe-the-esp32-boot-process-by-looking-at-the-serial-console

**Reply by final-option** [reply]
anscheinend holt die DTU weiterhin Daten, nur die GUI wird nicht angezeigt.

EDIT: hab gerade den DTU Tab neu geladen, und jetzt ist die GUI da. -> ???
Ok, das ist wirklich stranges Verhalten. Gibts über die GUI Zugriff auf logfiles die da vielleicht Licht ins Dunkel bringen würden?

Achja, ganz andere Frage: wo ist der Unterschied zum generic und generic_ESP32 bin file? Nur aus Interesse.

![grafik](https://github.com/tbnobody/OpenDTU/assets/138444296/b81de886-bfb0-4fa0-8a02-7d7387c550cd)

**Comment by tbnobody** [comment]
> Achja, ganz andere Frage: wo ist der Unterschied zum generic und generic_ESP32 bin file? Nur aus Interesse.

https://github.com/tbnobody/OpenDTU/wiki/Binary-Installation#overview-of-pre-compiled-binaries

> Gibts über die GUI Zugriff auf logfiles die da vielleicht Licht ins Dunkel bringen würden?

Das gerät ist ein MCU mit 4MB Flash und 340kb RAM... Also Log Files etc. oder auch SSH wird es darauf nicht geben.

Wenn es nun klappt könnte es auch ein Thema mit dem Browser Cache gewesen sein.

**Reply by final-option** [reply]
Moin.
Nochmal ein Update: 
nach dem Umstecken von Laptop auf Steckdose ging wieder nichts. Habe dann gesehen das der DTU AP aktiv war. Einloggen per Handy ging, die Zeitsynchronisation war weg, und die Netzwerkkonfig auch (SSID und PW aber noch da). 
Hab dann den Zeitserver von fritz.box auf pool.ntp.org umgestellt (ja, die Fritzbox ist eigentlich als Zeitserver konfiguriert, aber just in case) und noch mal gespeichert. Hat leider nichts gebracht, Netzwerkdaten wurden auch nicht angezeigt.
Erst als ich (weiterhin im DTU WLAN) aus der Oberfläche heraus einen Neustart gemacht habe wurden die Einstellungen übernommen, der DTU AP geschlossen und ich habe die GUI im Browser angezeigt bekommen.
Irgendwas ist (war) da krumm mit der dauerhaften Speicherung der Einstellungen, so sieht es zumindest für mich aus. 
Werde das mal beobachten ob der Fehler reproduzierbar ist.
Gruß

**Reply by Solarteur** [reply]
Eine Anmerkung zu "fritz.box" als Zeitserver: Läuft bei mir seit etwa einem Jahr völlig problemlos und störungsfrei.

**Reply by final-option** [reply]
Bei mir für den Rest des Netzwerkes auch. Ich wollte nur einfach ausschließen dass dies vielleicht das Problem (oder ein Teil davon) ist, da ja auch die Zeitdaten nicht gespeichert waren.

**Comment by tbnobody** [comment]
> da ja auch die Zeitdaten nicht gespeichert waren.

Was meinst du mit Zeitdaten gespeichert? Ein ESP kann keine Zeit speichern. Es gibt keine Hardware dafür. Die Zeit wird bei jedem Power Cycle vom Zeitserver geholt. Wenn keine WiFi Verbindung vorhanden ist --> Keine Uhrzeit.

**Reply by final-option** [reply]
Naja, ich hatte ja alle Einstellungen inklusive Zeitserver (fritz.box) gemacht im DTU WLAN am Laptop. Dort konnte ich dann ja auch die Livedaten sehen, die DTU hatte also auch die aktuelle Zeit. Dann auf WLAN umgestellt - zuerst nix, dann ja auf der seriellen Konsole geschaut wo ja Daten ankamen, und dann ging ja auch plötzlich die GUI wieder.
Danach vom laptop an die Steckdose, und wie oben beschrieben waren alle Einstellungen weg; ich musste mich erneut ins DTU WLAN einloggen, alle Einstellungen neu machen (Nur der WLAN Zugang war gespeichert) und die DTU neu starten.
Mit Zeitdaten weg meinte ich also, das nach dem Umstecken in die Steckdose die Zeit wieder auf 1.1.1970 stand.

**Comment by tbnobody** [comment]
>  Mit Zeitdaten weg meinte ich also, das nach dem Umstecken in die Steckdose die Zeit wieder auf 1.1.1970 stand.

Das ist normal. Der ESP verliert wie gesagt bei einem Power Cycle die Zeit. Diese wird erst vom NTP Server wieder geholt.

**Reply by final-option** [reply]
Ok, das hatte ich dann falsch verstanden und vermutet, dies könnte vielleicht mit zum Problem mit der GUI beigetragen haben. 
Dann habe ich aber aktuell auch keine Vermutung, was da schief läuft. Am Netzteil kann es eigentlich nicht liegen, habe 3 verschiedene getestet. Und es trat ja auch am Laptop auf - nur das es da dann (wohl eher zufällig) plötzlich wieder lief.

**Reply by final-option** [reply]
Muss mich leider noch mal wieder melden.
Nachdem es ja gestern an der Steckdose dann lief, heute wieder nicht. Habe nix geändert über Nacht.
GUI nicht erreichbar. Auf der Fritzbox geschaut, im Netzwerk aber mit 1Mbit - schonmal kein gutes Zeichen. 
Ok, Powercycle. Hat sich danach nicht mit dem WLAN verbunden, sondern der DTU AP war wieder aktiv. Unter Einstellungen-Netzwerk wie immer die SSID und das PW noch da, also Speichern, dann versucht sich die DTU ja ins WLAN einzuwählen.
Klappt nur leider gar nicht, war kurz drin und dann wieder weg.

Kann das ESP Board einen Hardwaredefekt haben? Allerdings kann ich mich ja über den DTU AP einloggen, und nach manueller Zeitsynchronisierung auch die Live Daten vom WR sehen. Und manchmal gehts auch im WLAN. Aber anscheinend nicht stabil.
Das ist irgendwie total frustrierend!

**Reply by final-option** [reply]
Moin.
Hat eigentlich sonst niemand diese Probleme? Aktuell ist die DTU überhaupt nicht nutzbar. :(
Soll ich vielleicht mal eine alte FW flashen? Oder testweise mal die AHOY DTU FW flashen?

**Reply by Solarteur** [reply]
Ich kann natürlich nur für mich sprechen: OpenDTU läuft bei mir (schon lange) stabil und ohne Probleme.

Hast Du schon einmal
- einen anderen Browser getestet?
- wie hier https://github.com/tbnobody/OpenDTU#flash-with-espressif-flash-download-tool-windows beschrieben OpenDTU über die Schritte "Erase" und dann "Start" unter Verwendung der **passenden** "factory.bin" neu geflasht und dann OpenDTU neu installiert?
- Kannst Du ein anderes ESP32-Modul testen - einen Hardware-Defekt hast Du ja selbst angesprochen?

**Reply by final-option** [reply]
Hi.
Ja, ich habe schon einmal mit der generic.factory.bin neu geflasht mit dem Espressif Tool. Dazu hatte ich weiter oben auch was geschrieben.
Nein, ich habe leider keine alternative Hardware zur Verfügung zum Testen.
Jein. Ich habe FireFox und Chrome probiert an 2 verschiedenen Laptops. Das Problem ist aber, dass sich die DTU bisher nur ein oder zwei mal mit der normalen Bandbreite mit der Fritzbox verbunden hat. Wie oben beschrieben ist es seitdem so, dass man die DTU (nachdem sie ihren eigenen AP abschaltet und sich mit dem WLAN verbindet) zwar in der Geräteliste als verbunden sieht, aber nur mit 1/1 Mbit für up und downstream. Und dann ist sie auch nicht im Browser erreichbar. Oft verschwindet sie dann auch nach einer Weile aus der Liste der verbundenen Geräte. 
Da die DTU an sich ja funktioniert - wenn ich mit dem DTU AP verbunden bin sehe ich ja die live Daten vom Wechselrichter, und in den Einstellungen sind auch die WLAN Daten zu sehen - gibts hier irgendwie ein Problem mit der Verbindung zum WLAN. 
Ich weiß nur nicht, wie ich es eingrenzen soll. DTU-seitig scheint es keine Möglichkeit in der Firmware zu geben, das Problem zu lokalisieren, da es keine Logfiles oder debug files gibt. Auf meiner Seite kann ich vielleicht noch mal das Gerät aus der Fritzbox löschen. 
Eventuell mal eine ältere FW flashen. Mehr fällt mir nicht ein, bin auch kein Programmierer.

**Comment by Gitbert23** [comment]
Hi,
bei mir war es der falsche Funkadapter  der keine Verbindung zum Wechselrichter hergestellt hat. Der cmt2300a wäre der richtige hatte aber den nrf24l01 der mit dem Wechselrichter nicht funktionierte.
Anfänger Fehler meinerseits.

😏

**Comment by Andre0be** [comment]
Hallo,

das Thema ist schon älter, für mich trotzdem aktuell. Der ESP32 hat Probleme mit WLAN und einer Fritzbox. Bei mir geht OpenDTU nur, wenn auf WLAN Kanal 1 gesendet wird. Es ist leider ein bekanntes Problem des ESP32. Bisher gibt es meines Wissens nach keine Lösung. Hatte jetzt noch das Problem als die FritzBox wieder Kanal 6 genommen hat, so das OpenDTU wieder nicht ging.

**Reply by Solarteur** [reply]
In dieser allgemeinen Form kann ich das mit Kanal 1 nicht bestätigen. Ich habe gerade "Autokanal" ausgeschaltet und OpenDTU mit Kanal 6 und Kanal 11 gestestet: Keine Probleme!

FritzBox 7530, FRITZ!OS: 7.57, WPA 2 + WPA 3

**Reply by Andre0be** [reply]
Dann scheint das Mesh das Problem zu sein. Mein OpenDTU ESP32 hängt an einem 1200 ax, der ist als AP konfiguriert und nicht als Repeater. Da funktioniert der ESP32 nur, wenn der 1200 ax mit Kanal 1 sendet.

**Reply by ghost** [reply]
Hi,
zum Thema kann ich auch nur Vermutungen anstellen, da @Andre0be hier das Mesh ansprach, könnte ich auch von kürzlichen eigenen Problemen berichten.
Meine FB7590 lauft mit dem Beta OS von AVM und darf auch selbstständig Updaten.
Ich habe 2 Repeater im Mesh, die als LAN-Brücke laufen.
Seit dem letzten Betaupdate sich mir gelegentlich diverse Probleme aufgefallen (chronologische Reihenfolge):
- Lappi meiner Frau geht nicht freiwillig ins WLAN: Ein Neustart des Repeaters in der Nähe half augenscheinlich.
- Handys sind mal im WLAN, mal nicht: Ein Aus-/ Einschalten des WLAN am Handy half.
- Nach dem letzten Update verbindet sich die openDTU schlecht mit dem WLAN: Ein Neustart der FB7590 half augenscheinlich.
- Rein zufällig habe ich einen Blick in die Netzwerk-Einstellungen meiner FB geschaut, die Repeater waren nicht mehr im Mesh, Ich denke auch, das die das selber gar nicht wussten.

Ich habe nicht darauf geachtet, mit welchen Gerät sich meine WLAN-Teilnehmer tatsächlich verbunden hatten. Über Kanäle habe ich nicht nachgedacht.
Das Mesh wurde wieder aufgebaut und ich habe bisher keine Probleme mehr bemerkt.
